sbndqm/dqmAnalysis/PMT/SBND/CAENV1730StreamsSBND_module.cc

```diff
@@ -0,0 +1,277 @@
+
+  int level = 3; 
+
+  artdaq::MetricMode mode = artdaq::MetricMode::Average;
+  artdaq::MetricMode rate = artdaq::MetricMode::Rate;
+
+  std::string groupName = "PMT";
+
+
+  // Now we look at the waveforms 
+  art::Handle< std::vector<raw::OpDetWaveform> > opdetHandle;
+  evt.getByLabel( m_opdetwaveform_tag, opdetHandle );
+
+  if( opdetHandle.isValid() && !opdetHandle->empty() ) {
+
+    // Create a sample with only one waveforms per channel
+
+    std::vector<unsigned int> m_unique_channels; 
+
+    for ( auto const & opdetwaveform : *opdetHandle ) {
+
+      if( m_unique_channels.size() > nTotalChannels ) { break; }
+
+      unsigned int const pmtId = opdetwaveform.ChannelNumber();
+      
+      auto findIt = std::find( m_unique_channels.begin(), m_unique_channels.end(), pmtId );
+
+      if( findIt != m_unique_channels.end() ) { continue; }
+		
+      // We have never seen this channel before
+      m_unique_channels.push_back( pmtId );
+      
+      std::string pmtId_s = std::to_string(pmtId);
+
+      int16_t baseline = Median( opdetwaveform );
+	
+      pulseRecoManager.Reconstruct( opdetwaveform );
+      std::vector<double>  pedestal_sigma = pedAlg->Sigma();
+        
+      double rms = Median( pedestal_sigma );
+
+      auto const& pulses = threshAlg->GetPulses();
+      double npulses = (double)pulses.size();
+
+      std::cout << pmtId << " " << baseline << " " << rms << " " << npulses << std::endl;
+
+      // Send the metrics 
+      sbndaq::sendMetric(groupName, pmtId_s, "baseline", baseline, level, mode); // Send baseline information
+      sbndaq::sendMetric(groupName, pmtId_s, "rms", rms, level, mode); // Send rms information
+      sbndaq::sendMetric(groupName, pmtId_s, "rate", npulses, level, rate); // Send rate information
+          
+
+      // Now we send a copy of the waveforms
+      double tickPeriod = 0.002; // [us]
+      std::vector<std::vector<raw::ADC_Count_t>> adcs {opdetwaveform};
+      std::vector<int> start { 0 }; // We are considreing each waveform independent for now
+
+      // send waveform
+      sbndaq::SendSplitWaveform("snapshot:waveform:PMT:" + pmtId_s, adcs, start, tickPeriod);
+
+
+      // Now we send a copy of the waveeform FFT
+      size_t NADC = opdetwaveform.Waveform().size();
+      double tickPeriodFFT = 1./tickPeriod; // [us], change this harcoding
+      //std::cout<<"Waveform size: "<<NADC<<" tickPeriodFFT="<<tickPeriodFFT<<std::endl;
+      _fft_manager.Set(NADC);
+      
+      // fill FFT
+      for (size_t i = 0; i < NADC; i ++) {
+	double *input = _fft_manager.InputAt(i);
+        *input = (double) opdetwaveform.Waveform().at(i);
+      }
+      _fft_manager.Execute();
+
+      std::vector<float> adcsFFT;
+      double real=0, im=0;
+      for(size_t k=0; k<_fft_manager.OutputSize();k++){
+        real=_fft_manager.ReOutputAt(k);
+        im=_fft_manager.ImOutputAt(k);
+        adcsFFT.push_back( std::sqrt(real*real+im*im) );
```

Comment:
Introducing `std::hypot()`:
```suggestion
        adcsFFT.push_back( std::hypot(real,im) );
```